src/Microsoft.AspNet.Razor/Parser/TagHelpers/TagHelperParseTreeRewriter.cs

```diff
@@ -207,6 +259,25 @@ private static bool ValidTagStructure(string tagName, Block tag, RewritingContex
             return !invalidStructure;
```

Comment:
nit: clearer to add `return false;` after the `OnError()` call and `return true;` here